### PR TITLE
refactor(deviation_evaluator): simplify plot function

### PR DESCRIPTION
**Pull request by kminoda**
Signed-off-by: kminoda <koji.minoda@tier4.jp>

## Description
I refactored deviation_evaluation_visualizer script for better readability.
The performance does not change before and after this PR except:
- `lower_bound` calculation (fixed a small mistake in previous version)
- duration-to-error plot (removed the plot, since it seemed unnecessary)

## Tests performed
Ran the tool for several data and confirmed that the results are valid

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kminoda** [comment]
Is it OK for me to request you a review for the modification in the python script...?

**Comment by kminoda** [comment]
@YamatoAndo こちらお時間のあるときにお願いします m(_ _)m

**Review by YamatoAndo — APPROVED** [review]

File: localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+                    msg.pose.pose.position.z,
+                    yaw,
+                ]
+            )
+            pose_cov_list.append(msg.pose.covariance.reshape(6, 6)[(0, 1, 5), :][:, (0, 1, 5)])
+        sorted_idxs = np.argsort(np.array(pose_list)[:, 0])
+        pose_list = np.array(pose_list)[sorted_idxs]
+        pose_cov_list = np.array(pose_cov_list)[sorted_idxs]
+        pose_list = np.array(pose_list)
+        pose_cov_list = np.array(pose_cov_list)
```
Comment (YamatoAndo) [inline localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py:130]: unnecessary lines

File: localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
```
Comment (YamatoAndo) [inline localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py]: unnecessary lines

File: localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py (hunk 1 above)
Comment (kminoda) [reply localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py:130]: Let me resolve this in the following PRs

**Comment by kminoda** [comment]
@kenji-miyake Sorry to bother you.
`pre-commit` GitHub Actions does not run somehow (it's skipped). Do you have any idea how to make it run so that this PR can be merged?

**Comment by kenji-miyake** [comment]
@kminoda It has run, but failed. Please fix them manually since some fixes aren't automatically applied.

**Comment by kminoda** [comment]
I see, thank you for the reply!

**Comment by kminoda** [comment]
@kenji-miyake I fixed the pre-commit warnings on the scripts I modified, but still the `pre-commit / pre-commit (pull_request)` is skipped. Do you know how to fix this?
Screenshot from 2022-11-01 18-13-38

**Comment by kenji-miyake** [comment]
@kminoda Ah, I see. It's intended to run only on private repositories.
For public repositories, please see `pre-commit.ci - pr` instead.

**Comment by kminoda** [comment]
@kenji-miyake OK! But another problem is that the `pre-commit / pre-commit (pull_request)` is set as a required workflow for merging the PR. Maybe we should set this as a non-mandatory workflow?

**Comment by kenji-miyake** [comment]
Exactly.
@yabuta Could you confirm the settings? :pray:

**Comment by yabuta** [comment]
@kminoda @kenji-miyake I changed this setting.

**Pull request merged**